src/client.ts

```diff
@@ -25,15 +25,142 @@ const ClientConfigurationSchema = Type.Object(
+   * // }
+   * ```
+   *
+   * @param indexName - The name of the index to describe.
+   * @returns A promise that resolves to {@link IndexMeta}
+   */
   describeIndex: ReturnType<typeof describeIndex>;
+
+  /**
+   * List all Pinecone indexes
+   * @example
+   * ```js
+   * const indexes = await client.listIndexes()
+   * console.log(indexes)
+   * // [ 'my-index', 'my-other-index' ]
+   * ```
+   *
+   * @returns A promise that resolves to an array of index names
+   */
   listIndexes: ReturnType<typeof listIndexes>;
+
+  /**
+   * Creates a new index.
+   *
+   * @example
+   * The minimum required configuration to create an index is the index name and dimension.
+   * ```js
+   * await client.createIndex({ name: 'my-index', dimension: 128 })
+   * ```
+   * @example
+   * In a more expansive example, you can specify the metric, number of pods, number of replicas, and pod type.
+   * ```js
+   * await client.createIndex({
+   *  name: 'my-index',
+   *  dimension: 128,
+   *  metric: 'cosine',
+   *  pods: 1,
+   *  replicas: 2,
+   *  podType: 'p1.x1'
+   * })
+   * ```
+   *
+   * @example
+   * By default all metadata fields are indexed when vectors are upserted with metadata, but if you want to improve performance you can specify the specific fields you want to index. This example is showing a few hypothetical metadata fields, but the values you'd use depend on what metadata you plan to store in Pinecone alongside your vectors.
+   * ```js
+   * await client.createIndex({ name: 'my-index', dimension: 128, metadataConfig: { 'indexed' : ['productName', 'productDescription'] }})
+   * ```
+   *
+   * @param options - The index configuration.
+   * @param options.name - The name of the index. Must be unique within the project and contain alphanumeric and hyphen characters. The name must start and end with alphanumeric characters.
+   * @param options.dimension - The dimension of the index. Must be a positive integer. The number you choose here will depend on the model you are using. For example, if you are using a model that outputs 128-dimensional vectors, you should set the dimension to 128.
+   * @param options.metric - The metric of the index. The default metric is `'cosine'`. Supported metrics include `'cosine'`, `'dotproduct'`, and `'euclidean'`. To learn more about these options, see [Distance metrics](https://docs.pinecone.io/docs/indexes#distance-metrics)
+   * @param options.pods - The number of pods in the index. The default number of pods is 1.
+   * @param options.replicas - The number of replicas in the index. The default number of replicas is 1.
+   * @param options.podType - The type of pod in the index. This string should combine a base pod type (`s1`, `p1`, or `p2`) with a size (`x1`, `x2`, `x4`, or `x8`) into a string such as `p1.x1` or `s1.x4`. The default pod type is `p1.x1`. For more information on these, see this guide on [pod types and sizes](https://docs.pinecone.io/docs/indexes#pods-pod-types-and-pod-sizes)
```

Comment:
just curious -- when you say this works with typescript, would the docs generation error if these param tags don't make sense with the actual function?

Reply:
By "works with typescript" I mainly mean that at a basic level it can compile code containing typescript annotations without erroring. I started out trying to use jsdoc which is a similar tool but much older which had a big hand in defining the comment syntax used; unfortunately jsdoc immediately errors when parsing source code containing typescript annotations. You could potentially rube goldberg something together with jsdoc by running it on the compiled `tsc` output. But that sounded not great, and doesn't leverage at all the information contained in the types. So I decided to look elsewhere.

I'm still getting to know this tool. It does seem to display type information when it is available, but I'm not sure yet if it would actually error if your written comment conflicted with the type def. So far I haven't seen full-on errors, but this thing emits a lot of different warnings that I need to study more.